api/v1alpha1/ndb_api_helpers.go

```diff
@@ -159,43 +160,151 @@ func GetNoneTimeMachineSLA(ctx context.Context, ndbclient *ndbclient.NDBClient)
 	return sla, fmt.Errorf("NONE TimeMachine not found")
 }
 
+// +kubebuilder:object:generate:=false
+type ProfileResolver interface {
+	resolve(ctx context.Context, allProfiles []ProfileResponse, filter func(allProfiles ProfileResponse) bool) (profile ProfileResponse, err error)
+}
+
+func (providedProfile *Profile) resolve(ctx context.Context, allProfiles []ProfileResponse, filter func(allProfiles ProfileResponse) bool) (profile ProfileResponse, err error) {
+	log := ctrllog.FromContext(ctx)
+	log.Info("resolving...", "provided profile", providedProfile)
+	// If both NAME and ID are not provided, return OOB profile
+	if providedProfile.Name == "" && providedProfile.Id == "" {
+		log.Info("Both Name and ID are empty")
+		profile, err = util.FindFirst(allProfiles, filter)
+
+		if err != nil {
+			log.Error(err, "Error resolving OOB Profile")
+			return profile, errors.New("no OOB profile found")
```

Comment:
Can we also add " and no profile information provided in the input" to the error?